torch_xla/distributed/xla_dist.py

```diff
@@ -86,32 +104,57 @@ def __init__(self,
                ' will interfere with the values set for distributed'
                ' training'.format(dist_var)))
 
+  def _check_client_mesh_health(
+      self, uneven_health_timeout=300, even_health_timeout=1800):
+    uneven = False
+    max_delay = 0
+    now = time.time()
+    master_hb = self._last_heartbeats[self._client_master.get_internal_ip()]
```

Comment:
IMHO it is confusing to differentiate the master. Look at my code snippet. Does not make any `master` assumption 😉

Reply:
I thought that assuming master was always alive would be a fine thing to do since if master is dead the `xla_dist` process would be dead anyways. But gotcha, updated to not differentiate at all.